### PR TITLE
Prevent Possible buffer overflow

### PR DESCRIPTION
**Pull request by dairyo**
Setting error message can be overflow.

### CODE REVIEW

**Comment by satoshikumano** [comment]
- Let's remove -std=gnu89 flag in make as we using snprintf which is not compatible with C89.
- Let's use snprintf instead of sprintf for other parts of the samples. (Even though the size is fixed, snprintf is preferable if environment allows to use it.)

**Comment by dairyo** [comment]
@satoshikumano I fixed according to your comments

> Let's remove -std=gnu89 flag in make as we using snprintf which is not compatible with C89.

This is fixed with 6aa80d2

> Let's use snprintf instead of sprintf for other parts of the samples. (Even though the size is fixed, snprintf is preferable if environment allows to use it.)

This is fixed with c485a72

**Comment by satoshikumano** [comment]
LGTM.
(Found that writer interface is confusing. Will deal with it in another PR.)

**Comment by dairyo** [comment]
@satoshikumano I added an issue in spread sheet. Can I merge this?

File: linux-sample/example.c [hunk 1]
```diff
@@ -207,24 +208,26 @@ static kii_bool_t state_handler(
...
+                smartlight.color[0], smartlight.color[1], smartlight.color[2]);
         if ((*writer)(kii, buf) == KII_FALSE) {
             return KII_FALSE;
         }
 
         if ((*writer)(kii, "\"colorTemperature\":") == KII_FALSE) {
             return KII_FALSE;
         }
-        sprintf(buf, "%d}", smartlight.color_temperature);
+        snprintf(buf, sizeof(buf) / sizeof(buf), "%d}",
```
Comment (satoshikumano) [inline linux-sample/example.c:229]: sizeof(buf[0])

**Pull request merged**